src/main/webapp/app/shared/sidebar/sidebar-card-medium/sidebar-card-medium.component.ts

```diff
@@ -0,0 +1,41 @@
+import { Component, Input } from '@angular/core';
+import { DifficultyLevel } from 'app/entities/exercise.model';
+import { SidebarCardElement, SidebarTypes } from 'app/types/sidebar';
+import { Subscription } from 'rxjs';
+import { SidebarEventService } from '../sidebar-event.service';
+import { ActivatedRoute, Router } from '@angular/router';
+@Component({
+    selector: 'jhi-medium-sidebar-card',
+    templateUrl: './sidebar-card-medium.component.html',
+    styleUrls: ['./sidebar-card-medium.component.scss'],
+})
+export class SidebarCardMediumComponent {
+    DifficultyLevel = DifficultyLevel;
+    @Input() sidebarItem: SidebarCardElement;
```

Comment:
I would set the sidebarItem required as it must not be null anyways.
```suggestion
    @Input({required:true}) sidebarItem: SidebarCardElement;
```